### PR TITLE
Rework script.

### LINKED ISSUE #12: Install apt-transport-https if needed
From [SRU comment #7](https://bugs.launchpad.net/ubuntu/+source/base-files/+bug/1686183/comments/7), the ubuntu-advantage script should install the apt-transport-https package if it's not installed, so that apt can access the ESM archive which is behind https.

### PR DESCRIPTION
**Pull request by albertodonato**
- copy the keyring file instead of importing with apt-key.
- install apt-transport-https if not installed.
- rework tests

Closes #12 and #13

### CODE REVIEW

File: ubuntu-advantage [hunk 1]
```diff
@@ -71,9 +16,16 @@ deb https://${1}@${REPO_URL}/ubuntu precise main
...
+is_package_installed() {
+    dpkg-query -s "$1" >/dev/null 2>&1
+}
+
 
 enable_esm() {
-    import_gpg_key
+    cp "${KEYRINGS_DIR}/${REPO_KEY_FILE}" "$APT_KEYS_DIR"
+    if ! is_package_installed apt-transport-https; then
+        apt-get install apt-transport-https
```
Comment (panlinux) [inline ubuntu-advantage]: I think we should use `apt-get install -y` here. As an improvement, we could wrap this install in a nice message saying "Installing missing dependency <foo>..." and only display apt's output in the case of an error, or else the screen gets too polluted in my opinion: see https://pastebin.canonical.com/187004/

File: ubuntu-advantage [hunk 2]
```diff
@@ -82,6 +34,7 @@ enable_esm() {
 disable_esm() {
     if [ -f "$REPO_LIST" ]; then
         mv "$REPO_LIST" "${REPO_LIST}.save"
+        rm  "$APT_KEYS_DIR/$REPO_KEY_FILE"
```
Comment (panlinux) [inline ubuntu-advantage]: This `rm` should be guarded with `-f`, or the script can fail if that file doesn't exist by any chance:

```
ubuntu@precise-esm:~$ sudo rm /etc/apt/trusted.gpg.d//ubuntu-keyring-esm.gpg
ubuntu@precise-esm:~$ sudo ubuntu-advantage disable-esm
rm: cannot remove `/etc/apt/trusted.gpg.d/ubuntu-keyring-esm.gpg': No such file or directory
ubuntu@precise-esm:~$ 
```

File: ubuntu-advantage [hunk 3]
```diff
@@ -71,9 +16,16 @@ deb https://${1}@${REPO_URL}/ubuntu precise main
 EOF
 }
 
+is_package_installed() {
+    dpkg-query -s "$1" >/dev/null 2>&1
```
Comment (panlinux) [inline ubuntu-advantage]: Suggestion: `dpkg-query -W` has less output to get rid of than `-s`:
```
root@precise-esm:~# dpkg-query -W base-files; echo $?
base-files	6.5ubuntu6.9
0
root@precise-esm:~# dpkg-query -W asdasdasd; echo $?
No packages found matching asdasdasd.
1
root@precise-esm:~# 
```

File: ubuntu-advantage (hunk 2 above)
Comment (panlinux) [inline ubuntu-advantage]: Should we run `apt-get update` here perhaps?

Comment (dpb1) [inline ubuntu-advantage]: I'd rather not do an apt-get update.  perhaps... print a message.

Comment (albertodonato) [inline ubuntu-advantage]: yeah, that's a corner case (you have no reason to remove the file yourself), but fixed

File: ubuntu-advantage (hunk 3 above)
Comment (albertodonato) [inline ubuntu-advantage]: Changed

File: ubuntu-advantage (hunk 2 above)
Comment (albertodonato) [inline ubuntu-advantage]: we do print a message when enabling and disabling telling the use to run `apt-get update`

File: ubuntu-advantage (hunk 3 above)
Comment (albertodonato) [inline ubuntu-advantage]: Actually, in precise `-W` will print the package name (without version) if it's not installed. So back to `-s`

File: ubuntu-advantage (hunk 1 above)
Comment (albertodonato) [inline ubuntu-advantage]: Done

**Comment by albertodonato** [comment]
Closes #11 too

**Comment by panlinux** [comment]
Shipit

**Pull request merged**